### PR TITLE
Feat: Slack Bot Tool

### PR DESCRIPTION
**Pull request by manthanguptaa**

### CODE REVIEW

**Comment by anuragts** [comment]
LGTM

File: phi/tools/slack.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
+import json
+
+from slack_sdk import WebClient
+from slack_sdk.errors import SlackApiError
+
```
Comment (ashpreetbedi) [inline phi/tools/slack.py:5]: wrap in try/except Import error. see: https://github.com/phidatahq/phidata/blob/main/phi/tools/duckdb.py#L6-L9

File: phi/tools/slack.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+from phi.utils.log import logger
+
+
+class SlackTools(Toolkit):
+    def __init__(self, token: str):
+        super().__init__(name="slack")
+        self.client = WebClient(token=token)
+        self.register(self.send_message)
+        self.register(self.list_channels)
+        self.register(self.get_channel_history)
```
Comment (ashpreetbedi) [inline phi/tools/slack.py:16]: wrap in variables eg:

if send_message:
    ...
    
see: https://github.com/phidatahq/phidata/blob/main/phi/tools/duckdb.py#L36-L45

**Review by ashpreetbedi — CHANGES_REQUESTED** [review]
Really, really good PR. Just requested small changes to the structure but everything else looks perfect

**Review by ashpreetbedi — APPROVED** [review]

**Pull request merged**